app/services/student_removal_service.rb

```diff
@@ -1,42 +1,63 @@
 # frozen_string_literal: true
 
 class StudentRemovalService
-  def initialize(students:, school:, remove_from_profile: false, token: nil)
-    @students = students
+  class NoSchoolError < StandardError; end
+  class NoClassesError < StandardError; end
+  class StudentHasProjectsError < StandardError; end
+  class NoopError < StandardError; end
+
+  def initialize(school:, remove_from_profile: false, token: nil, raise_on_noop: false)
```

Comment:
I guess a fundamental question is why is this asking for a school when the service is a "student removal service"?  I'd suggest maybe renaming it to `SchoolStudentsRemovalService` or let it handle student IDs directly, and pass them in one at a time.

Reply:
but yes, that's good call about altering the name - that'll help make it clearer